src/ORM/Connect/PDOQuery.php

```diff
@@ -18,13 +47,70 @@ class PDOQuery extends Query
 
     protected $results = null;
 
+    /**
+     * @var string
+     */
+    protected $fetchMedthod = self::FETCH_METHOD_ALL;
+
+    /**
+     * Global fetch method setting
+     * @var string
+     */
+    protected static $fetchMethodFlag = self::FETCH_METHOD_ALL;
+
+    /**
+     * @param string $method
+     */
+    public static function setFetchMethodFlag($method)
+    {
+        if (!in_array($method, [static::FETCH_METHOD_ALL, static::FETCH_METHOD_ROW])) {
+            return;
+        }
+
+        static::$fetchMethodFlag = $method;
+    }
+
+    /**
+     * @return string
+     */
+    public static function getFetchMethodFlag()
+    {
+        return static::$fetchMethodFlag;
+    }
+
+    /**
+     * Invoke a callback which may modify fetch method flag, but ensures this flag is restored
+     * after completion, without modifying global state.
+     *
+     * The desired fetch method flag should be set by the callback directly
+     *
+     * @param callable $callback
+     * @return mixed Result of $callback
+     */
+    public static function withFetchMethodFlag($callback)
```

Comment:
Wouldn't it make sense to also ask for the intended fetch method as a parameter so this method can handle both changing to a new method and back to the existing method?